### PR TITLE
[fix](routine load) fix ROUTINE LOAD bug,kafka commit a lack of one(#17282)

### LINKED ISSUE #17282: [Bug] 在使用 ROUTINE LOAD 导入时，commit kafka 时会少一位 offset
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.2

### What's Wrong?

在使用ROUTINE LOAD 导入时，导入数据正常，数据库的数据对的上。但是显示在 kafka 处会显示一条未消费
企业微信截图_1677227058949

### What You Expected?

kafka 的消费记录正常

### How to Reproduce?

在 1.2 中使用 ROUTINE LOAD kafka 导入

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yuxuan-luo**
# Proposed changes

Issue Number: close #17282 

## Problem summary

base on f2a96fd commit test
https://github.com/apache/doris/commit/f2a96fda62117412141e13e8df4a1f7589163956

## Checklist(Required)

* [ X] Does it affect the original behavior
* [ X] Has unit tests been added
* [ X] Has document been added or modified
* [ X] Does it need to update dependencies
* [ √] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 1]
```diff
@@ -320,8 +320,8 @@ void RoutineLoadTaskExecutor::exec_task(std::shared_ptr<StreamLoadContext> ctx,
 
         std::vector<RdKafka::TopicPartition*> topic_partitions;
         for (auto& kv : ctx->kafka_info->cmt_offset) {
-            RdKafka::TopicPartition* tp1 =
-                    RdKafka::TopicPartition::create(ctx->kafka_info->topic, kv.first, kv.second);
```
Comment (yiguolei) [inline be/src/runtime/routine_load/routine_load_task_executor.cpp:324]: Could you please explain why?

**Comment by yiguolei** [comment]
./run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.83 seconds
 stream load tsv:          481 seconds loaded 74807831229 Bytes, about 148 MB/s
 stream load json:         40 seconds loaded 2358488459 Bytes, about 56 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230302060838_clickbench_pr_106887.html

File: be/src/runtime/routine_load/routine_load_task_executor.cpp (hunk 1 above)
Comment (yuxuan-luo) [reply be/src/runtime/routine_load/routine_load_task_executor.cpp:324]: The offsets you commit are the offsets of the messages you want to read next (not the offsets of the messages you did read last).

**Comment by yiguolei** [comment]
Is this only a display problem or it will lost data in the past?

**Comment by yuxuan-luo** [comment]
> 这只是一个显示问题还是会丢失过去的数据？
only a display problem

**Comment by caiconghui** [comment]
once change the commit offset, you should also change consume logic for commit offset, or you may cause data lost. what's more, it would be not compatible with old verison doris, which means it would cause one data lost when you first upgrade doris cluster.

**Comment by caiconghui** [comment]
once change the commit offset, you should also change consume logic for commit offset, or you may cause data lost. what's more, it would be not compatible with old verison doris, which means it would cause one data lost when you first upgrade doris cluster.

**Comment by yiguolei** [comment]
If this is only a display problem, I think we'd better not modify it.

**Comment by morningman** [comment]
> If this is only a display problem, I think we'd better not modify it.

This is the right way to commit the kafka offset to kafka server, So it LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — DISMISSED** [review]

**Comment by yuxuan-luo** [comment]
run clickbench

**Comment by yuxuan-luo** [comment]
run P0 Regression

**Comment by yuxuan-luo** [comment]
run P0 Regression

**Comment by yuxuan-luo** [comment]
run P0 Regression

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Pull request merged**